server/pbenchinacan/container-build.sh

```diff
@@ -37,19 +40,21 @@ fi
 # Container build section.
 #-
 
-# Open a copy of the base container.  Docker format is required in order to set
-# the hostname.
-container=$(buildah from --format=docker ${BASE_IMAGE})
+# Open a copy of the base container.
+container=$(buildah from ${BASE_IMAGE})
 
 buildah config \
     --label maintainer="Pbench Maintainers <pbench@googlegroups.com>" \
     $container
 
 buildah copy $container ${RPM_PATH} /tmp/pbench-server.rpm
 buildah run $container dnf update -y
-buildah run $container dnf install -y --setopt=tsflags=nodocs \
+if [[ "${BASE_IMAGE}" == *"ubi9:latest" ]]; then
+    buildah run $container dnf install -y --nodocs \
         https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
-buildah run $container dnf install -y /tmp/pbench-server.rpm nginx less rsyslog rsyslog-mmjsonparse
+fi
+buildah run $container dnf install -y --nodocs \
+    /tmp/pbench-server.rpm nginx less rsyslog rsyslog-mmjsonparse
```

Comment:
This fails when I run it on my laptop:
```
+ buildah run ubi9-working-container-2 dnf install -y --nodocs /tmp/pbench-server.rpm nginx less rsyslog rsyslog-mmjsonparse
Updating Subscription Management repositories.
Unable to read consumer identity
Subscription Manager is operating in container mode.

This system is not registered with an entitlement server. You can use subscription-manager to register.

Extra Packages for Enterprise Linux 9 - x86_64                                                          4.7 MB/s |  13 MB     00:02    
Last metadata expiration check: 0:00:03 ago on Mon Feb  6 15:54:30 2023.
No match for argument: rsyslog-mmjsonparse
Error: Unable to find a match: rsyslog-mmjsonparse
Error: while running runtime: exit status 1
```

Reply:
You need to use `BASE_IMAGE=fedora:37`, or something other than RHEL 9 to install outside of the build system method.

Our CI jobs use special images which can't be used outside of that environment, something that needs another PR or set of PRs to correct.

Reply:
See the first commit, https://github.com/distributed-system-analysis/pbench/pull/3211/commits/833288083f1e9ed0cd4ee41421c7d1406e254890.

Reply:
Once I got past the two stumbling blocks I mentioned [below](https://github.com/distributed-system-analysis/pbench/pull/3211#discussion_r1097751732), I ran into the same issue that @ndokos did.

How is this working for the CI but not for us?  (I _think_ the CI is using the default base image, which is what I think I'm using.)

Reply:
Near as I can figure, the Jenkins servers have a valid RHEL license, so using the UBI containers on those systems works.

Reply:
> the Jenkins servers have a valid RHEL license, so using the UBI containers on those systems works.

It's unfortunate that the new dependency on `rsyslog-mmjsonparse` prevents us from doing builds on our VMs.

I tried using a Fedora-36 base image (because I had an RPM lying around for it), but that's not happy, either:
```
$ BASE_IMAGE=quay.io/fedora/fedora:36 RPM_PATH=[...] PB_CONTAINER_REG=[...] bash -xe ./server/pbenchinacan/container-build.sh

+ buildah run fedora-working-container-13 bash -c 'sed -i -e '\''/pam_loginuid/ s/^/#/'\'' /etc/pam.d/crond'
sed: can't read /etc/pam.d/crond: No such file or directory
Error: while running runtime: exit status 2
```
Do we need more conditionalization for non-RHEL platforms?

Reply:
Works fine with `jenkins/runlocal` for me.  Not sure what is missing.

Reply:
On my Fedora 37 box, if I do the following it works:

$ make -C server/rpm clean rpm
$ export BASE_IMAGE=fedora:37
$ echo "images.paas.redhat.com/pbench" > ~/.config/pbench/ci_registry.name
$ export RPM_PATH=~/rpmbuild/RPMS/noarch/pbench-server-*.rpm
$ server/pbenchinacan/container-build.sh

Reply:
Yeah, I think the `ci_registry.name` is key; without that, we run into a missing definition for `PB_CONTAINER_REG`.

Reply:
@ndokos, I have pushed an update that now removes the need to define `BASE_IMAGE`, and makes the RPM version built strictly match the base image that will be used.

Reply:
@ndokos, the following command line now works for me:
```
BASE_IMAGE=fedora:36 RPM_PATH=~/rpmbuild/RPMS/noarch/pbench-server-1.0.0-*.rpm PB_CONTAINER_REG=images.paas.redhat.com ./server/pbenchinacan/container-build.sh
```
It does not work if I omit the `BASE_IMAGE` definition or set it to the UBI-9 image, but I think we've flogged that horse sufficiently for now.  Hopefully, this resolves your concern.

(@portante, just to note, Nick's original difficulty was with an un-wrapped invocation of `container-build.sh`, but, unless I missed something, your update affects only invocations of `jenkins/runlocal`.)

Reply:
I did `BASE_IMAGE=fedora:37 jenkins/runlocal` and it did indeed pass - thanks!